Use v0.25.0 api

Senyoret1 opened the pull request:
Fixes #160 

Changes:
- The connections to the node now use many of the improvements added in v0.25.0, Mainly the `verbose` argument and the ability to send long lists of addresses using POST.
- A way to determine the version of the node was created. This is currently used to maintain compatibility with the version 0.24.0 of the node and to use the new features only if the node is runnin v0.25.0 or higher. 
- The function responsible for sending POST requests was updated, to be able to send more than Json.
- The unit tests were updated to be compatible with the changes.

mahansky commented on src/app/services/api.service.ts:
```diff
@@ -34,21 +34,39 @@ export class ApiService {
     if (requestCsrf) {
```
`requestCsrf` is never `true`, this could be removed

Senyoret1 commented:
Compatibility with previous versions is maintained for several reasons. One of the main ones is to make it easily to update the wallet and the public node without downtime: if the wallet is compatible with both versions, it is possible to update the wallet before the public node and everything will work correctly both before and after the public node is updated. Another reason, even more important, is that the wallet is compatible with multiple coins and all coins could be running a different version of the node (in fact in the future some coins could be several version behind compared to Skycoin).

As you suggested, I simplified the version checking code a bit. However, it is not optimal to simply save the node version in a variable and consult it, because the version of the node is obtained asynchronously after changing the coin or manually changing the URL of the node. The main problem is that when changing the coin the code starts to perform operations that need to know the version of the node, but the variable still contains the version of the node of the previously selected coin. The approach used in this pr allows to ensure that the version of the node is readed only after having updated it, to avoid the previous problem. Using `BehaviorSubject` with an initial value of null instead of `ReplaySubject` is part of the approach, since null is used to indicate that the version of the node has not yet been updated (the value returns to be null when needed, like after changing the coin).

I created a new issue with the parts that should use BigNumber: https://github.com/skycoin/skycoin-web/issues/486

I also deleted the code related to `requestCsrf`, as you recommended.

gz-c commented:
For version comparison, use the X.YY in vX.YY.ZZ, ZZ is for bugfix releases and won't affect the API

Senyoret1 commented:
The code that makes the comparison takes into account the Z value, just in case. However, the code is only checking if the version is 0.25.0 or higher, so in practice the Z value is not being taking into account. The "rc" part is ignored.

gz-c commented:
The Z part needs to be ignored in case we have to do a bugfix release, we won't want to interfere with the light client behavior

Senyoret1 commented:
I modified the function and now the Z part is always ignored
